packages/loader/container-loader/src/container.ts

```diff
@@ -1275,13 +1264,30 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         const snapshotTree = getSnapshotTreeFromSerializedContainer(detachedContainerSnapshot);
         this._storage.loadSnapshotForRehydratingContainer(snapshotTree);
         const attributes = await this.getDocumentAttributes(this._storage, snapshotTree);
-        assert(attributes.sequenceNumber === 0, 0x0db /* "Seq number in detached container should be 0!!" */);
+
+        // Needs driver definitions. To be resolved via #9833.
+        const acceptsNonZeroSeqStartNum: boolean =
+            (this.serviceFactory as any).capabilities
+                ?.acceptsNonZeroSeqStartNum === true;
+        assert(
```

Comment:
Asserts should be used for internal invariants. Here we deal with condition that we do not control, so it should be an exception with reasonable message. This feels like UsageError to me.
I'd move it into the driver when file is created (driver that does not support that condition) and remove capability object.

Reply:
See the earlier comment. Container loader is not checking for non-zero sequence numbers any further.  Driver is now throwing generic error.  I'm not sure if we are intentionally associating layers with error types as `UsageError` seems to be used at loader-layer (and above). @markfields curious on your take here (around `UsageError` type?